### ISSUE TITLE
scattergl fill - autorange and restyle

### ISSUE THREAD

**Issue by alexcjohnson** [plotly/plotly.js#2354]
`scattergl` doesn't match the `scatter` svg autorange behavior with fill: for example, `fill: 'tozeroy'` like in the `gl2d_axes_labels2` mock should trigger the y axis to autorange to zero, and the x axis to autorange tightly rather than padded, despite the markers.

Also: open the same mock and `Plotly.restyle(gd, {fill: 'none'})` - the fill will not disappear, though if you *add* a fill to a plot that doesn't have one, it *will* appear.

**Comment by etpinard** [comment]
> Also: open the same mock and Plotly.restyle(gd, {fill: 'none'}) - the fill will not disappear, though if you add a fill to a plot that doesn't have one, it will appear.

cc @dfcreative 

-----

`scattergl` probably suffers from other autorange bugs and/or discrepancies.

**Comment by etpinard** [comment]
> Plotly.restyle(gd, {fill: 'none'}) - the fill will not disappear

this one is fixed via https://github.com/plotly/plotly.js/pull/2377. The autorange issues prevails.

**Comment by etpinard** [comment]
@dfcreative I found a pretty bad typo in the scattergl calc step where the marker options container is refered to `options.markers` [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L110), [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L114), [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L115) and [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L121) instead of `options.markers` as set [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L513).

This means that `Axis.expand` never got the correct size values [here](https://github.com/plotly/plotly.js/blob/1256c4f9de37ad68ba82d28a048dbe96bc204c4d/src/traces/scattergl/index.js#L134-L135) and the autorange computation for scattergl were plain wrong. See image diffs in commit https://github.com/plotly/plotly.js/commit/58c29724b7daa7211830245af2645f256401310a .

In short, I'll spend some time during to clean up the scattergl calc, by hopefully reusing as much of `Scatter.calc` as possible.

**Comment by etpinard** [comment]
done in #2404

**Comment by etpinard** [comment]
Related: https://github.com/plotly/plotly.js/issues/2417
